### PR TITLE
[corlib] System.Text.Encoding.* from CoreFX

### PR DESCRIPTION
**Pull request by EgorBo**
Second part of https://github.com/mono/mono/issues/7247
Needs https://github.com/mono/corefx/pull/25 to be merged.

All tests from corefx were already ported to mono.

### CODE REVIEW

File: mcs/build/common/SR.cs [hunk 1]
```diff
@@ -46,7 +46,18 @@ internal static string Format(string resourceFormat, object p1, object p2, objec
 		return string.Format (CultureInfo.InvariantCulture, resourceFormat, p1, p2, p3);
 	}
 
-	internal static string GetResourceString (string str) => str;
+	internal static string GetResourceString (string str)
+	{
+		foreach (var field in typeof (SR).GetFields ())
```
Comment (marek-safar) [inline mcs/build/common/SR.cs]: Why do we need it? (we don't want reflection code in mscorlib)

Comment (EgorBo) [reply mcs/build/common/SR.cs]: @marek-safar agree. But what should I do with https://github.com/dotnet/corefx/blob/43d07018fd0c4176b20481e958bcd1ccadacc23a/src/Common/src/CoreLib/System/Text/Encoding.cs#L450 - it wants to get a proper name for encoding based on Resources

Comment (EgorBo) [reply mcs/build/common/SR.cs]: I can make a switch for this case

**Comment by EgorBo** [comment]
@monojenkins build deb with monolite

**Comment by marek-safar** [comment]
@monojenkins build

**Comment by EgorBo** [comment]
@monojenkins build deb with monolite

**Comment by EgorBo** [comment]
@monojenkins build deb with monolite

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by EgorBo** [comment]
@monojenkins build Linux x64

**Comment by marek-safar** [comment]
@EgorBo there are few failing tests and we also need to clean up `TypeForwardedFromAttribute`

File: mcs/class/corlib/corlib.dll.sources [hunk 2]
```diff
@@ -829,7 +829,6 @@ System.Security.Principal/WindowsIdentity.cs
 System.Security.Principal/WindowsImpersonationContext.cs
 System.Security.Principal/WindowsPrincipal.cs
 System.Text/EncodingHelper.cs
-System.Text/NormalizationForm.cs
```
Comment (marek-safar) [inline mcs/class/corlib/corlib.dll.sources]: We should delete the file

File: mcs/class/corlib/corlib_xtest.dll.sources [hunk 3]
```diff
@@ -124,33 +124,20 @@
...
-../../../external/corefx/src/System.Text.Encoding/tests/Encoding/*.cs
-../../../external/corefx/src/System.Text.Encoding/tests/Fallback/*.cs
-../../../external/corefx/src/System.Text.Encoding/tests/Latin1Encoding/*.cs
-../../../external/corefx/src/System.Text.Encoding/tests/UnicodeEncoding/*.cs
-../../../external/corefx/src/System.Text.Encoding/tests/UTF7Encoding/*.cs
-../../../external/corefx/src/System.Text.Encoding/tests/UTF8Encoding/*.cs
-../../../external/corefx/src/System.Text.Encoding/tests/UTF32Encoding/*.cs
-../../../external/corefx/src/System.Text.Encoding/tests/EncodingTestHelpers.cs
-../../../external/corefx/src/System.Text.Encoding/tests/NegativeEncodingTests.cs
+#../../../external/corefx/src/System.Text.Encoding/tests/ASCIIEncoding/*.cs
```
Comment (marek-safar) [inline mcs/class/corlib/corlib_xtest.dll.sources]: Why are all being disabled?

File: scripts/ci/run-test-default.sh [hunk 4]
```diff
@@ -45,11 +45,6 @@ ${TESTCMD} --label=System.Runtime.Remoting --timeout=5m make -w -C mcs/class/Sys
 ${TESTCMD} --label=Cscompmgd --timeout=5m make -w -C mcs/class/Cscompmgd run-test;
 ${TESTCMD} --label=Commons.Xml.Relaxng --timeout=5m make -w -C mcs/class/Commons.Xml.Relaxng run-test;
 ${TESTCMD} --label=System.ServiceProcess --timeout=5m make -w -C mcs/class/System.ServiceProcess run-test
-${TESTCMD} --label=I18N.CJK --timeout=5m make -w -C mcs/class/I18N/CJK run-test
```
Comment (marek-safar) [inline scripts/ci/run-test-default.sh]: These tests should not be affected

**Comment by EgorBo** [comment]
the code still relies on old Encoding+I18N. Changing it to corefx Encoding+System.Text.CodePages will require additional work in Xamarin.IOS, Xamarin.Android and linker.

**Comment by EgorBo** [comment]
@monojenkins build

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins commit apidiff and csproj

**Comment by marek-safar** [comment]
@monojenkins commit apidiff

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff and csproj

**Pull request merged**